azurerm_postgresql_flexible_server_configuration - restart server for static parameters

neil-yechenwei opened the pull request:
Service team raised a request to restart server after user sets the static parameters. So I submitted this PR to support it.

![image](https://user-images.githubusercontent.com/19754191/212805642-a283cc95-7144-4e0d-9f86-b864add70fd5.png)

jackofallops reviewed (changes requested):
hi @neil-yechenwei 
Thanks for this PR.  It looks mostly fine to me, but could you add a test that covers the update path you've added to ensure values are updated successfully and as expected? 

Thanks!

neil-yechenwei commented:
@jackofallops , thanks for your comment. Actually I already added a test case in this PR to cover the logic I added since "cron.max_running_jobs" is static parameter.

![image](https://user-images.githubusercontent.com/19754191/213056640-92b75055-3ca5-4102-97d2-01f367d2ff52.png)

katbyte reviewed (approved):
LGTM 🏷️
